google-oauth-client/src/main/java/com/google/api/client/auth/oauth/OAuthHmacSigner.java

```diff
@@ -32,14 +34,34 @@
 @Beta
 public final class OAuthHmacSigner implements OAuthSigner {
 
+  private final Map<String, String> signatureMethodMap = ImmutableMap.of(
+          "HMAC-SHA1", "HmacSHA1",
+          "HMAC-SHA256", "HmacSHA256"
+  );
+
+  private String signatureMethod = "HMAC-SHA1";
+
   /** Client-shared secret or {@code null} for none. */
   public String clientSharedSecret;
 
   /** Token-shared secret or {@code null} for none. */
   public String tokenSharedSecret;
 
+  private String getAlgorithm() {
+    return signatureMethodMap.get(signatureMethod);
+  }
+
   public String getSignatureMethod() {
-    return "HMAC-SHA1";
+    return signatureMethod;
+  }
+
+  /** Set the signature method.  Valid signature methods are "HMAC-SHA1" and "HMAC-SHA256" */
+  public void setSignatureMethod(String signatureMethod) throws IllegalArgumentException {
+    if (signatureMethodMap.containsKey(signatureMethod)) {
```

Comment:
Maybe use an enum instead?

Reply:
this has been reverted back in favor of a new class